### ISSUE TITLE
Review source code and files in repository

### ISSUE THREAD

**Issue by ghost** [kamilion/customizer#129]
This is the main initiative to prepare future release 4.2.0. The initiative has already started since I have made initial commit in tagged release 4.1.2.

**Justification**

It would be pointless to do this initiative, once the repository is transferred to future contributor and whenever migration to Python 3 and PyQt5 takes place. This initiative is indeed tedious, but someone has to do this anyway since most people will avoid this kind of "wasteful" efforts.

**What to do**

The job is basically **checking file by file** and determine whether the contents are compliant to respective specifications. **Then, fix it when necessary.** Even better, consider improving some files to be more useful and simple to update based on good examples found in other projects.

The review details shall be noted in each commit and perhaps in release notes.

**How to review**

All existing files in `master` branch can be found by clicking on 'Find file' button in home page, which links to File Finder at https://github.com/clearkimura/Customizer/find/master

Finally, when finishing 4.2.0 and no more files being created, list down each files at comment below with review date and status for confirmation.

**These "wasteful" efforts must be finished in future release 4.2.0.**

---

**Main references**
1. [Debian New Maintainers' Guide](https://www.debian.org/doc/manuals/maint-guide/), esp. Chapter 4. Required files under the **`debian`** directory, Chapter 7.4 Using **`lintian`**.
2. [Debian Policy Manual ](https://www.debian.org/doc/debian-policy/) esp. Chapter 7.2 Binary Dependencies.
3. [Debian Menu System](https://www.debian.org/doc/packaging-manuals/menu.html/) esp. Chapter 3 - The `menu` file.

**Comment by clearkimura** [comment]
Looking at many other projects, I suppose files at top level should be renamed to all UPPERCASE i.e. COPYING and the files that contain Markdown styling should have `.md` file extension i.e. README.md.

**Proposed changes to top level files**

1. `CHANGELOG.md`: Rename from existing `ChangeLog`; Adding file extension to the file will automatically convert special text i.e. commit number and issue number, which is convenient for allowing to visit relevant links quickly.

2. `LICENSE`: Not created yet; This will contain those hardcoded text in existing "License" tab. This may contain "For full text, refer to COPYING" to indicate a copy of full text also co-exist in the repository. No markdown styling; This file will be used by QTextBrowser, then show in GUI via Qt resource.

3. `Contributors`: Remove all Markdown styling; This will contain only contributors' full name and nickname (username) only. Copyright and ownership years may be added here instead of existing "License" tab. This file will also be used by QTextBrowser, then show in GUI via Qt resource.

4. `README.md`: Not sure yet; Need to see good example of README and make comparison; Remove duplicates and reduce amount of text to read; Much less than what has been written for Wiki.

5. `genabout.py`: Remove this file because no longer relevant, since we are changing approach to simple listing that doesn't require translation at all.

**Practice for copyright and ownership years**

A common practice is to place the copyright details in **About** tab/dialog of GUI, but will make difficult to update the years. This is because GUI label might require additional tool i.e. Qt Designer to edit the GUI label properly. By including the years as part of GUI, editing will be less accessible.

![about-copyright-1](https://cloud.githubusercontent.com/assets/1223304/18794890/5bc89b46-81f4-11e6-880d-edbe3c475082.png)

Next common practice is to include **Copyright** tab/dialog of GUI, then include additional buttons for respective FOSS licenses (In case of having multiple licenses).

![about-copyright-2](https://cloud.githubusercontent.com/assets/1223304/18794896/64262038-81f4-11e6-8aa1-530b332be22c.png)

While these are common practices, that doesn't mean we should follow exactly what other project does. We need something simple so that future repo owner can update easily without all the hassles i.e. of having to use additional tool to edit GUI, just to add the copyright years and name.

**Where to see copyright and ownership years**

For this project, I found copyright years loitering at few locations within the source code:

* `debian/copyright` (This is visible within package, but not elsewhere; This should remain to follow the standard according to Debian Policy Manual)

* `README.md` (This is visible in repository; This should remain because easy to notice and most projects commonly note in this file)

* `src/gui.ui` (This is within hardcoded text in source file) -- **Will be removed**

* `tr/*.ts` files (This is within hardcoded text in translation file for `gui.ui`) -- **Will be removed**

* `Contributors` (This is proposed location that will be both visible in repository and GUI; also more proper location instead of showing in `LICENSE` file or "License" tab)

**Justification for copyright and ownership years**

Similar to translating license is discouraged, I suppose owners' copyright years shouldn't be "mixed" together with legal statements of FOSS license (We include a copy of legal statements that should be preserved). This is why copyright and ownership years is more proper to include in `Contributors` file, which will be shown in **About** > **Credits** in GUI instead.

Both `debian/copyright` and `README.md` files are almost invisible after installation. The proposed changes will introduce a third location as alternative in `Contributors` file, which will be visible even after installation and will be shown via QTextBrowser in GUI.

To take common practice into consideration, I suppose we can add generic notes about copyright details in **About** tab of GUI. The generic notes may be something like this: 

> Customizer is copyright by Ivalio Monev, et al. For more details, see /usr/share/doc/customizer/copyright".

This is the good point of generic notes; Mention the original author and provide reference to existing document. No need to update this content again; Write properly at once, then forget.

**Additional notes**

Depending on project scale and community, some projects hardly mention the copyright and ownership years anywhere within the repository. If ever mentioned, it would be in README.md and together with license name, under COPYRIGHT section.

**TL;DR** The proposed changes will reduce duplicates and will be easier to update. For copyright and ownership years, include generic notes in **About** tab of GUI. For future repo owners, just add (c) YYYY-YYYY after the owner name in `Contributors` file and similarly in `debian/copyright` file. Both files can be edited using any text editor.

**The proposed changes for placing copyright years shall be reflected in #125 later.**

**Comment by ghost** [comment]
This issue shall be closed, because "old stable" will be deprecated.

Most of "wasteful" efforts were made against supporting files (not-so-related to core features) and documentation files. Skipped boring check of pylint. That's it.
